deployment/src/main/java/io/quarkus/jsch/deployment/JSchDevServiceConfig.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkus.jsch.deployment;
+
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
+
+@ConfigMapping(prefix = "quarkus.jsch.devservice")
+@ConfigRoot(phase = ConfigPhase.BUILD_TIME)
+public interface JSchDevServiceConfig {
+
+    /**
+     * Enable the JSch DevService.
+     */
+    @WithDefault("true")
```

Comment:
Perhaps it should be defaulted to `false` to avoid spinning up a service when it's not needed?

Reply:
That's fine I think. My only concern is about spinning up images when they are not needed, but that's okay I guess

Reply:
there is a `reuse` property which is set to `false` by default, I can set it to `true`.
the reason why i have choosed to set the `reuse` to `false` by default is:

* when `reuse` is `false`, the **ryuk** auto remove the container
* when `reuse` is `true`, TestContainer does not run **ryuk** so, the container is never stop and removed.
  To me, this use case is usefull for containers which need lots of resources and when we work with multiple services

Reply:
It would be nice to have that explanation added to the docs so we don't lose it

Reply:
when set `reuse` to `true`, unfortunately NO
It's the same behavior for official extension like rabbitmq client

Reply:
About the `reuse` flag, I think it should be set to `false` by default. Spinning up a new SSH server is fast and the devservice should IMHO live as long as the app is running

Reply:
Even better: it could be true but the devservice is only started if an injection point is found AND no host is configured for the session

Reply:
Context examples:

* Historical use of [Axway](https://docs.axway.com/bundle/TransferCFT_36_UsersGuide_allOS_en_HTML5/page/Content/administration/sftp/sftp_intro.htm#use_cases_) to transfer files from server to many servers
  The named session annotation for simple use cases
  And `JSchSessions.fromName(String sessionName)` method for dynamic use cases
  
  Due to the interop interface contracts between french telecom operators, the usage of Axway is mandatory for lots of applications

* S3 to SFTP, SFTP to SFTP: create simple and dynamic config at runtime with environment variables

Reply:
we can also create a new extension that depends on this one. (ex: `quarkus-jsch-simple`)
then we can keep the low level